cohortextractor/emis_backend.py

```diff
@@ -504,9 +504,11 @@ def get_date_condition(self, table, date_expr, between):
         min_date, max_date = between
         min_date_expr, join_tables1 = self.date_ref_to_sql_expr(min_date)
         max_date_expr, join_tables2 = self.date_ref_to_sql_expr(max_date)
+
+        base_join_table = base_join_table or table
         joins = [
             f"LEFT JOIN {join_table}\n"
-            f"ON {join_table}.registration_id = {table}.registration_id"
+            f"ON {join_table}.registration_id = {base_join_table}.registration_id"
```

Comment:
This gets the right results in the test, but I'm not confident whether keeping the LEFT JOIN and just replacing the table it's joining on it the right type of join to do